packages/app/src/specs/InlineSpecListTree.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <ul
+    ref="rootEl"
+    tabindex="0"
+  >
+    <li
+      v-for="(row, idx) in filteredTree"
+      :key="idx"
+      v-bind="rowProps"
+      class="
+        flex
+        outline-none
+        group
+        cursor-pointer
+        relative
+        before:(absolute
+        w-8px
+        left-[-4px]
+        h-28px
+        border-r-4 border-r-gray-1000
+        rounded-lg)
+        item
+      "
+      :style="{ paddingLeft: `${(row.depth - 2) * 10 + 16}px` }"
+      :class="{'before:hover:(transitional-all duration-250 ease-in-out border-r-indigo-300) before:focus:(border-r-indigo-300)': row.isLeaf, 'before:border-r-indigo-300': isCurrentSpec(row.data)}"
+      @click="onRowClick(row, idx)"
+      @keypress.enter.space.prevent="onRowClick(row, idx)"
+    >
+      <SpecFileItem
+        v-if="row.isLeaf"
+        :file-name="row.data?.fileName || row.value"
+        :extension="row.data?.specFileExtension || ''"
+        :selected="isCurrentSpec(row.data)"
+        class="ml-22px"
+      />
+      <DirectoryItem
+        v-else
+        :directories="row.value.split('/')"
+        :expanded="row.expanded.value"
+      />
+    </li>
+  </ul>
+</template>
+
+<script setup lang="ts">
+import type { FoundSpec } from '@packages/types'
+import { useCollapsibleTree, UseCollapsibleTreeNode } from '@packages/frontend-shared/src/composables/useCollapsibleTree'
+import { useListNavigation } from '@packages/frontend-shared/src/composables/useListNavigation'
+import { buildSpecTree, SpecTreeNode } from '@packages/frontend-shared/src/utils/buildSpecTree'
+import SpecFileItem from './SpecFileItem.vue'
+import { computed, ref, Ref } from 'vue'
+import DirectoryItem from './DirectoryItem.vue'
+import { useRouter } from 'vue-router'
+import { useSpecStore } from '../store'
+
+const props = defineProps<{
+  specs: FoundSpec[]
+}>()
+
+const router = useRouter()
+
+const specStore = useSpecStore()
+
+const isCurrentSpec = (spec: FoundSpec | undefined) => {
+  if (!spec) return false
+
+  return spec.relative === specStore.activeSpec?.relative
+}
+
+const specTree = computed(() => buildSpecTree(props.specs))
+const collapsible = computed(() => useCollapsibleTree(specTree.value, { dropRoot: true }))
+const filteredTree = computed(() => collapsible.value.tree.filter(((item) => !item.hidden.value)))
```

Comment:
It seems `specTree` and `collapsible` is only used by `filteredTree`. That means we create three computed properties, but really we only want one? Should this be:

```ts
const filteredTree = computed(() => {
  const specTree = buildSpecTree(props.specs))
  const collapsible = useCollapsibleTree(specTree.value, { dropRoot: true }))
  collapsible.value.tree.filter(((item) => !item.hidden.value))
})
```

Also, I don't think `buildSpecTree` should return an object with a `value` property. It makes it looks like it's returning a `ref` (because of the `.value` property). I was expecting to see a return type of `Ref` - but looking at the function, it's just a regular JS object with a non-reactive property that happens to be called `value`, unless I'm missing something?

Reply:
I tried this but it breaks the collapsing functionality. I believe it's because the computed properties referenced from `collapsible` are defined inside another computed property so Vue doesn't properly track the changes

Reply:
Yeah I noticed when I tried the refactor... This computed in a computed is a bit weird, I wonder if there's a better way.